WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
Some options

1. Add a dummy "zeroth" round to the arrays.

2. Use key-value semantics rather than arrays.

3. Have a library/class of utility functions for reading and writing to WCIF documents. They can be made correctly once and for all.

4. Go all the way and wrap the WCIF doc in an object that has proper functions to access the data.